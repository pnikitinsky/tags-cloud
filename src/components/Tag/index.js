export { default as Component } from './Tag';
