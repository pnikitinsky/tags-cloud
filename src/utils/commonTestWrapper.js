import React from 'react';
import renderer from 'react-test-renderer';
import reducers from 'src/reducers';
import { createStore } from 'redux';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { ThemeProvider } from 'styled-components';
import { createTheme } from 'src/styles/themes';

const store = createStore(
  reducers,
);

const commonTestWrapper = (children) => {
  return renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <ThemeProvider theme={createTheme()}>
          {children}
        </ThemeProvider>
      </MemoryRouter>
    </Provider>
  );
};

export default commonTestWrapper;
