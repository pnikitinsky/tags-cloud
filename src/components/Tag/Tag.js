import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { getDesiredTag } from 'src/utils/tags';
import { Component as Header } from 'src/components/Header';
import { Link } from 'react-router-dom';
import { MainContent } from 'src/screens/TagsScreen'

const Container = styled(MainContent)`
  color: white;
  align-items: center;
`;

const InfoItem = styled.li``;

const ItemsContainer = styled.ul`
  display: flex;
  flex-basis: 45%;
  flex-direction: column;
  justify-self: stretch;
  list-style-type: none;
`;

const BackToHomeLink = styled(Link)`
  display: inline-block;
  border: 1px solid orange;
  padding: 10px;
  color: white;
`;

class Tag extends React.Component {
  static propTypes = {
    match: PropTypes.object.isRequired,
  };

  componentDidMount() {
    const { match } = this.props;
    const { tagId } = match.params;
    if (tagId) {
      this.setState({
        desiredTagData: getDesiredTag(tagId),
      });
    }
  }

  state = {
    /**
     * @desiredTagData { object }, has to contain su data as:
     *   - @label { string }
     *   - @id { string }
     *   - @totalMentions { integer }
     *   - @negativeMentions { integer }
     *   - @positiveMentions { integer }
     *   - @neutralMentions { integer }
     *   - @pageTypes { object }
     */
    desiredTagData: {},
  };

  parsePageTypes = types => {
    return Object.entries(types).map(
      (type, key) => (
        <InfoItem key={key}>
          {
            type.toString()
          }
        </InfoItem>
      )
    );
  };

  render() {
    const { desiredTagData } = this.state;
    return (
      <Container>
        <Header screen={'tag'}/>
        {
          desiredTagData &&
          <ItemsContainer>
            <InfoItem>{`Label: ${desiredTagData.label}`}</InfoItem>
            <InfoItem>{`Total Mentions: ${desiredTagData.totalMentions || 0 }`}</InfoItem>
            <InfoItem>{`Negative Mentions: ${desiredTagData.negativeMentions || 0 }`}</InfoItem>
            <InfoItem>{`Positive Mentions: ${desiredTagData.positiveMentions || 0 }`}</InfoItem>
            <InfoItem>{`Neutral Mentions: ${desiredTagData.neutralMentions || 0 }`}</InfoItem>
            <ItemsContainer>
              <p>Page Types:</p>
              {
                desiredTagData.pageTypes &&
                this.parsePageTypes(desiredTagData.pageTypes)
              }
            </ItemsContainer>
          </ItemsContainer>
        }
        <BackToHomeLink to={'/tags'}>
          Back to the Home screen
        </BackToHomeLink>
      </Container>
    );
  }
}


export default Tag;
