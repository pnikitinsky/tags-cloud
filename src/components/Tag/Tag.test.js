import React from 'react';
import Tag from './Tag';
import commonTestWrapper from 'src/utils/commonTestWrapper';

test('Tag should match snapshot', () => {
  const mockedProps = {
    match: {
      params: {
        tagId: '1751295897__D_B Presents'
      }
    },
  };

  const component = commonTestWrapper(
    <Tag {...mockedProps}/>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
