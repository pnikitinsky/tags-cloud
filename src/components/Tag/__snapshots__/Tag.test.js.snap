// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag should match snapshot 1`] = `
<div
  className="sc-EHOje jmkGjr sc-bwzfXH DRIRY"
>
  <h1
    className="sc-bdVaJa ejDcSq"
  >
    Tag's Screen
  </h1>
  <ul
    className="sc-gzVnrw cnXzHt"
  >
    <li
      className="sc-bZQynM bOrPFB"
    >
      Label: D/B Presents
    </li>
    <li
      className="sc-bZQynM bOrPFB"
    >
      Total Mentions: 3
    </li>
    <li
      className="sc-bZQynM bOrPFB"
    >
      Negative Mentions: 0
    </li>
    <li
      className="sc-bZQynM bOrPFB"
    >
      Positive Mentions: 0
    </li>
    <li
      className="sc-bZQynM bOrPFB"
    >
      Neutral Mentions: 3
    </li>
    <ul
      className="sc-gzVnrw cnXzHt"
    >
      <p>
        Page Types:
      </p>
      <li
        className="sc-bZQynM bOrPFB"
      >
        blog,3
      </li>
      <li
        className="sc-bZQynM bOrPFB"
      >
        facebook,0
      </li>
      <li
        className="sc-bZQynM bOrPFB"
      >
        forum,0
      </li>
      <li
        className="sc-bZQynM bOrPFB"
      >
        general,0
      </li>
      <li
        className="sc-bZQynM bOrPFB"
      >
        image,0
      </li>
      <li
        className="sc-bZQynM bOrPFB"
      >
        news,0
      </li>
      <li
        className="sc-bZQynM bOrPFB"
      >
        review,0
      </li>
      <li
        className="sc-bZQynM bOrPFB"
      >
        twitter,0
      </li>
      <li
        className="sc-bZQynM bOrPFB"
      >
        video,0
      </li>
    </ul>
  </ul>
  <a
    className="sc-htoDjs bPMUhx"
    href="/tags"
    onClick={[Function]}
  >
    Back to the Home screen
  </a>
</div>
`;
